### PR TITLE
Flatpak packaging base

### PR DESCRIPTION
**Pull request by cra0zy**
Flatpak is a type of installer for Linux.

This PR provides an automated system that allows game made with .NET Core to be automatically packaged up into a flatpak when `dotnet publish` command is run.

https://www.nuget.org/packages/MonoGame.Packaging.Flatpak/

### CODE REVIEW

**Comment by tomspilman** [comment]
Can we put the MonoGameTeam copyright headers in these files?

**Comment by cra0zy** [comment]
> Can we put the MonoGameTeam copyright headers in these files?

Oh, right, sorry. I had my own set since I was submitting a prerelease version to nuget for now.

**Comment by cra0zy** [comment]
PS. The way you build this is with the following command:
```
msbuild /p:Configuration=Release /t:Pack
```

This will generate a nuget package from the project.

**Comment by tomspilman** [comment]
> PS. The way you build this is with the following command:

Should we get this added to the automated build then?

**Comment by cra0zy** [comment]
> Should we get this added to the automated build then?

Not until we remove Protobuild... this only works with .NET Core version of DesktopGL.

**Comment by tomspilman** [comment]
I see.

File: Tools/MonoGame.Packaging.Flatpak/BuildFlatpakTask.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
+using System;
```
[the remark this answers is not in the document]
Comment (cra0zy) [reply Tools/MonoGame.Packaging.Flatpak/BuildFlatpakTask.cs]: Done.

Note to self, submit PR when less sleepy.

**Comment by tomspilman** [comment]
Thanks @cra0zy !

**Pull request merged**